cryostatio/cryostat
# feat(archives): allow target-specific archived recording uploads with jvmId parameter

# Welcome to Cryostat! 👋
## Before contributing, make sure you have:
_______________________________________________

Fixes: #1301 

## Description of the change:
Allows recordings to be uploaded to specific target archives from jvmId. Also factors some methods out of `RecordingsPostHandler` and other constant Strings.

## Motivation for the change:
See #1301 

## How to manually test:
1. Run PR normally with smoketest.
2. Check jvmId -> `JVM_ID` of some target. e.g. vertx-fib-demo-1 (cryostat:9093) from TargetSelect dropdown (won't work on targets that need auth -> will be null <-- this needs fixing)
3. have some archived recording relative path saved -> `RECORDING_PATH`
4. Run
```bash
$ https -f --verify=no -v POST :8181/api/beta/recordings/JVM_ID"Authorization: Basic dXNlcjpwYXNzCg==" recording@RECORDING_PATH labels='{"foo": "bar"}'
```
5. Notice recording is uploaded and with labels as well.
6. (There is some front-end bug with duplicate recordings? Refreshing page, fixes this)
